crates/oci/src/client.rs

```diff
@@ -164,12 +160,27 @@ impl Client {
         locked.components = components;
         locked.metadata.remove("origin");
 
-        let oci_config = Config {
-            data: serde_json::to_vec(&locked)?,
-            media_type: SPIN_APPLICATION_MEDIA_TYPE.to_string(),
-            annotations: None,
+        // Push layer for locked spin application config
+        let locked_config_layer = ImageLayer::new(
+            serde_json::to_vec(&locked).context("could not serialize locked config")?,
+            SPIN_APPLICATION_MEDIA_TYPE.to_string(),
+            None,
+        );
+        layers.push(locked_config_layer);
+
+        // Construct empty/default OCI config file. Data may be parsed according to
+        // the expected config structure per the image spec, so we want to ensure it conforms.
+        // (See https://github.com/opencontainers/image-spec/blob/main/config.md)
+        // TODO: Explore adding data applicable to the Spin app being published.
+        let oci_config_file = ConfigFile {
+            architecture: oci_distribution::config::Architecture::Wasm,
+            os: oci_distribution::config::Os::Wasip1,
+            ..Default::default()
```

Comment:
One minor but user experience would be to provide a default entry point https://github.com/containerd/runwasi/blob/2fb77889d5d7b0e590a574cfbc44089b54036811/crates/oci-tar-builder/src/bin.rs#L61C39-L61C39

if this isn't set then when running with ctr need to provide one: 



which is the `shim.wasm` in the command `sudo ./ctr run --net-host --rm --runtime=io.containerd.spin.v1 docker.io/
jsturtevant/spin-wasm-shim:latest-2.0 testwasm shim.wasm`

otherwise end up with 

```
time="2023-10-30T22:55:30.029486688Z" level=error msg="on non-Windows, at least one process arg entry is required"

time="2023-10-30T22:55:30.029555687Z" level=error msg="failed to initialize container process: missing args in the process spec"
```

Reply:
@jsturtevant -- what exactly would be the meaning for the "default entrypoint" for a Spin app, which can have multiple components (which means no _one_ wasm component)?

Reply:
This value is passed as arg0 in wasmtime shim.  Maybe it isn't used spin shim?

Reply:
My assumption is that it wouldn't be used in the Spin shim, as the information about components is part of the locked app manifest.

Reply:
Yes, I don't think it is blocking.  I was using the applications name in runwasi. I am afk, but can look what happens in the docker container scenario later

Reply:
Spin currently ignores this value, runwasi parses it and sends to the shim via `RuntimeContext` but it is ignored by spin's shim: https://github.com/deislabs/containerd-wasm-shims/blob/6ca8df8247d94cf1dc6ce5e529b00ee728aee4ec/containerd-shim-spin-v1/src/engine.rs#L123

The reason this is working when packages as a container is because `/` is the command is set on the yaml: https://github.com/deislabs/containerd-wasm-shims/blob/6ca8df8247d94cf1dc6ce5e529b00ee728aee4ec/README.md?plain=1#L95

In the case of a shim such as wasmtime/wasmedge it makes sense to have pass the entrypoint info as `arg0` and we even do some additional parsing to allow for entering other functions in the wasm module https://github.com/containerd/runwasi/blob/642cafacde77d25762c8c0c0ca78ff1010caff16/crates/containerd-shim-wasm/src/container/context.rs#L17-L25

So in this case we can continue to use `/` in the command for the pod yaml or you could set this value in the config (maybe image name?) and users wouldn't need to set the command to `/` in the yaml.  In the future maybe it can be useful otherwise we can revisit this in the next iteration of the config support in containerd.